deploy/stacks/lambda_api.py

```diff
@@ -150,6 +157,64 @@ def __init__(
                 )
             )
 
+        if custom_auth is not None:
+            # Create the custom authorizer lambda
+            custom_authorizer_assets = os.path.realpath(
+                os.path.join(
+                    os.path.dirname(__file__),
+                    '..',
+                    'custom_resources',
+                    'custom_authorizer',
+                )
+            )
+
+            custom_lambda_env = {
+                'envname': envname,
+                'LOG_LEVEL': 'DEBUG',
+                'custom_auth_provider': custom_auth.get('provider'),
+                'custom_auth_url': custom_auth.get('url'),
+                'custom_auth_client': custom_auth.get('client_id'),
+                'custom_auth_jwks_url': custom_auth.get('jwks_url')
+            }
+
+            for claims_map in custom_auth.get('claims_mapping', {}):
+                custom_lambda_env[claims_map] = custom_auth.get('claims_mapping', '').get(claims_map, '')
+
+            self.authorizer_fn = _lambda.Function(
+                self,
+                f'CustomAuthorizerFunction-{envname}',
+                function_name=f'{resource_prefix}-{envname}-custom-authorizer',
+                handler='custom_authorizer_lambda.lambda_handler',
+                code=_lambda.Code.from_asset(
+                    path=custom_authorizer_assets,
+                    bundling=BundlingOptions(
+                        image=_lambda.Runtime.PYTHON_3_9.bundling_image,
+                        local=SolutionBundling(source_path=custom_authorizer_assets),
+                    ),
+                ),
+                memory_size=512 if prod_sizing else 256,
+                description='dataall Custom authorizer replacing cognito authorizer',
+                timeout=Duration.seconds(20),
+                environment=custom_lambda_env,
+                vpc=vpc,
```

Comment:
Why are we not creating a security group for the Lambda? I am not a security expert, but I imagine a scenario where our Lambda is called by anyone in the internet and might be throttled (being very paranoid-minded)

Reply:
Yes. I think you are right. Added a security groups similar to other Lambdas

Reply:
When a security group was not specifically mentioned, the lambda creation attached a security group by default with no inbound and outbound rules. Although, this takes care of having a security group but I have explicitly added security group for this custom authorizer lambda

Reply:
I looked into the docs. A Lambda can be deployed without a VPC or in a VPC. If deployed in a VPC, it always has security groups, but as you pointed out, better to have control over the rules. I have check the new implementation, we can resolve this comment